pkg/snapstore/s3_snapstore.go

```diff
@@ -540,6 +556,19 @@ func (s *S3SnapStore) Delete(snap brtypes.Snapshot) error {
 
 // GetS3CredentialsLastModifiedTime returns the latest modification timestamp of the AWS credential file(s)
 func GetS3CredentialsLastModifiedTime() (time.Time, error) {
+	// TODO: @renormalize Remove this extra handling in v0.31.0
+	// Check if a JSON file is present in the directory, if it is present -> the JSON file must be used for credentials.
+	if dir, isSet := os.LookupEnv(awsCredentialDirectory); isSet {
+		modificationTimeStamp, err := getJSONCredentialModifiedTime(dir, "AWS")
+		if err != nil {
+			return time.Time{}, err
+		}
+		if !modificationTimeStamp.IsZero() {
+			return modificationTimeStamp, nil
+		}
+		// Non JSON credential files might exist in the credential directory, do not return
+	}
+
```

Comment:
If you follow my above suggestion then here you don't need extra handling of this, may be you need to just swap the code .... check `JSON` first then `non-JSON`